content/blog/managing-kubernetes-infrastructure-with-dotnet-and-pulumi/index.md

```diff
@@ -0,0 +1,251 @@
+
+Using .NET to build our Kubernetes infrastruture offers several benefits:
+* Strong Typing:  Unlike YAML, C# and F# offer a rich type system with quick feedback on potential errors.  
+* Rich IDE Support:  Use the rich features of IDEs like Visual Studio and Visual Studio Code to develop you Kubernetes infrastructure - completion lists, refactorings, intellisense and more.   
+* Familiar Languages and APIs:  Apply all the features of C#, F# and VB.NET to your Kubernetes infrastructure - loops, variables, and the entire ecosystem of .NET Core libraries from `System` to everything in Nuget.
+* Components and Classes:  Instead of copy/pasting pages of YAML between projects, .NET code can abstract common functionality into classes and libraries for code re-use and clean infrastructure design.  
+
+Together, these provide an entirely more familiar experience for working with Kubernetes than using YAML or Helm (a mix of YAML and Go templates) for .NET developers.
+
+# Tour of Kubernetes with .NET
+
+We can see a simple example in practice - deploying an Nginx pod into a Kubernetes cluster.
+
+```cs
+var pod = new Pod("pod", new PodArgs
+{
+    Spec = new PodSpecArgs
+    {
+        Containers =
+        {
+            new ContainerArgs
+            {
+                Name = "nginx",
+                Image = "nginx",
+            },
+        },
+    },
+});
+```
+
+We can deploy this with `pulumi up`.
+
+```
+$ pulumi up
+Previewing update (luke-dev):
+
+     Type                    Name                              Plan       
+ +   pulumi:pulumi:Stack     basic_dotnet_kubernetes-luke-dev  create     
+ +   └─ kubernetes:core:Pod  pod                               create     
+ 
+Resources:
+    + 2 to create
+
+Do you want to perform this update? yes
+Updating (luke-dev):
+
+     Type                    Name                              Status      
+ +   pulumi:pulumi:Stack     basic_dotnet_kubernetes-luke-dev  created     
+ +   └─ kubernetes:core:Pod  pod                               created     
+ 
+Resources:
+    + 2 created
+
+Duration: 11s
+
+Permalink: https://app.pulumi.com/lukehoban/basic_dotnet_kubernetes/luke-dev/updates/81
+```
+
+As always, Pulumi programs describe the desired state of our infrasttructure, instead of an imperative process to construct that infrastructure.  This means that if we change our program, Pulumi will compute the minimal delta to apply to our Kubernetes cluster to transition to this new desired state.  This almost feel like using Edit-and-Continue on our deployed Kubernetes resources - modifying our Kubernetes resources in place inside the cluster.
```

Comment:
```suggestion
As always, Pulumi programs describe the desired state of our infrastructure, instead of an imperative process to construct that infrastructure.  This means that if we change our program, Pulumi will compute the minimal delta to apply to our Kubernetes cluster to transition to this new desired state.  This almost feels like using Edit-and-Continue on our deployed Kubernetes resources---modifying our Kubernetes resources in place inside the cluster.
```